File: python/selection_sort.py
#frist example for selection sort

#it should return min number index in list
def select(nums: list, start: int) -> int:
	length = len(nums)
	pos = start
	for i in range(start, length):
		if nums[i] < nums[pos]:
			pos = i

	return pos

def selection_sort(nums: list) -> None:
	for i in range(len(nums)):
		pos = select(nums, i)
		nums[i], nums[pos] = nums[pos], nums[i]


#next example for selection sort
def findIndex(nums: list) -> int:
	length = len(arr)
	pos = 0
	for i in range(1, length):
		if nums[i] < nums[pos]:
			pos = i

	return pos

def selectionSort(nums: list) -> list:
	newList = []

	for i in range(len(nums)):
		pos = findIndex(nums)
		newList.append(nums.pop(pos))

	return newList

File: javascript/selection_sort.js
//frist example for selection sort

// it should return number index
function select(nums, start){
	let length = nums.length;
	let minIndex = start;

	for (let i = start; i < length; i++){
		if(nums[i] < nums[minIndex]){
			minIndex = i;
		}
	}

	return minIndex;
}

function selectionSort(nums) {
	for(let i = 0; i < nums.length; i++){
		let minIndex = select(nums, i);
		let temp = nums[minIndex];
		nums[minIndex] = nums[i];
		nums[i] = temp;
	}
}


//next example

function findIndex(nums) {
	let minIndex = 0;

	for(let i = 0; i < nums.length; i++){
		if(nums[i] < nums[minIndex]){
			minIndex = i
		}
	}

	return minIndex;
}

function swapIndex(nums) {
	let length = nums.length;
	let newNums = []
	for(let i = 0; i < length; i++){
		let minIndex = findIndex(nums);
		newNums.push(nums.splice(minIndex, 1)[0]);
	}

	return newNums;
}
